### PR TITLE
[BUGFIX] Fix Boom shootable line specials not being repeatable

### PR DESCRIPTION
**Pull request by electricbrass**
Additionally removes the check for the line to have a tag to address #991. Different ports vary with respect to this behavior, with Odamex itself being inconsistent between the different trigger types, so I'm open to reverting this if desired or keeping the change.

### CODE REVIEW

File: common/p_boomfspec.cpp [hunk 1]
```diff
@@ -3306,8 +3306,8 @@ bool P_ShootCompatibleSpecialLine(AActor* thing, line_t* line)
 			if (line->flags & ML_SECRET) // they can't open secret doors either
 				return false;
 		}
-		if (!line->id) // e6y //jff 3/2/98 all gun generalized types require tag
-			return false;
+		// if (!line->id) // e6y //jff 3/2/98 all gun generalized types require tag
```
Comment (fuzztooth) [inline common/p_boomfspec.cpp]: Can these just be removed rather than commented out?

Comment (electricbrass) [reply common/p_boomfspec.cpp]: I commented it out to match all the other commented out ones. Wasn't sure if there was a reason for them being like that.

Comment (electricbrass) [reply common/p_boomfspec.cpp]: Throughout this file there's a lot of nearly identical checks, some commented out, some not. So it seems like a larger cleanup of this is needed.

Comment (electricbrass) [reply common/p_boomfspec.cpp]: I've removed it and the other commented out checks, as well as the remaining uncommented checks in the same function.

**Review by fuzztooth — APPROVED** [review]

**Pull request merged**